packages/snaps-controllers/src/test-utils/controller.ts

```diff
@@ -85,6 +89,21 @@ export class MockApprovalController {
     return promise;
   }
 
+  hasRequest(
+    opts: { id?: string; origin?: string; type?: string } = {},
+  ): boolean {
+    return this.#approval?.request.id === opts.id;
+  }
```

Comment:
If we only use the ID to check if there is a request, this function could only take the ID:

```suggestion
  hasRequest(
    id?: string,
  ): boolean {
    return this.#approval?.request.id === opts.id;
  }
```